osquery/tables/system/linux/portage.cpp

```diff
@@ -0,0 +1,326 @@
+      r["build_time"] = buildCont;
+    }
+
+    std::string repoCont;
+    if (readFile(directory + "/repository", repoCont).ok()) {
+      boost::trim(repoCont);
+      r["repository"] = repoCont;
+    }
+
+    results.push_back(r);
+  }
+
+  return results;
+}
+
+QueryData parsePortageKeywordSummaryContent(const std::string& keywords,
+                                            const std::string& masked,
+                                            const std::string& unmasked) {
+  QueryData results;
+  std::unordered_map<std::string, PortagePackage*> tempStorage;
+
+  if (keywords.length() > 0) {
+    for (const auto& i : osquery::split(keywords, "\n")) {
+      auto line = split(i);
+      if (line.size() == 0 || boost::starts_with(line[0], "#")) {
+        continue;
+      }
+      PortagePackage* r = new PortagePackage();
+      std::pair<std::string, std::string> package =
+          portageSplitPackageVersion(line[0]);
+      r->package = package.first;
+      r->version = package.second;
+      if (line.size() > 1) {
+        r->keyword = line[1];
+        boost::algorithm::trim_all(r->keyword);
+      }
+      r->mask = "0";
+      r->unmask = "0";
+      if (!boost::starts_with(r->keyword, "#")) {
+        tempStorage.emplace(std::make_pair(line[0], r));
+      }
+    }
+  }
+
+  if (masked.length() > 0) {
+    for (const auto& i : osquery::split(masked, "\n")) {
+      auto line = split(i);
+      if (line.size() == 0 || boost::starts_with(line[0], "#")) {
+        continue;
+      }
+
+      std::unordered_map<std::string, PortagePackage*>::iterator it =
+          tempStorage.find(line[0]);
+      if (tempStorage.end() != it) {
+        it->second->mask = "1";
+      } else {
+        PortagePackage* r = new PortagePackage();
+        std::pair<std::string, std::string> package =
+            portageSplitPackageVersion(line[0]);
+
+        r->package = package.first;
+        r->version = package.second;
+        r->keyword = "";
+        r->mask = "1";
+        r->unmask = "0";
+        tempStorage.emplace(std::make_pair(line[0], r));
+      }
+    }
+  }
+
+  if (unmasked.length() > 0) {
+    for (const auto& i : osquery::split(unmasked, "\n")) {
+      auto line = split(i);
+      if (line.size() == 0 || boost::starts_with(line[0], "#")) {
+        continue;
+      }
+
+      std::unordered_map<std::string, PortagePackage*>::iterator it =
+          tempStorage.find(line[0]);
+      if (tempStorage.end() != it) {
+        it->second->unmask = "1";
+      } else {
+        PortagePackage* r = new PortagePackage();
+        std::pair<std::string, std::string> package =
+            portageSplitPackageVersion(line[0]);
+
+        r->package = package.first;
+        r->version = package.second;
+        r->keyword = "";
+        r->mask = "0";
+        r->unmask = "1";
+        tempStorage.emplace(std::make_pair(line[0], r));
+      }
+    }
+  }
+
+  for (auto it = tempStorage.begin(); it != tempStorage.end(); ++it) {
```

Comment:
If using a `shared_ptr` or move semantics for the second item, this can take the normal range-based iterator loop form.